resources/js/components/UseExpandable.vue

```diff
@@ -0,0 +1,139 @@
+<script>
+
+import { ref, computed, onMounted } from 'vue'
+
+export default function useExpandable(itemSection, items, initialItem, setHash = true, wrap = true, vertical = false) {
+
+    const activeItem = ref('')
+    const currentFocusItem = ref('')
+
+    const itemKeys = computed(() => { return Object.keys(items)})
+    const firstItemKey = computed(() => { return itemKeys.value[0] })
+    const lastItemKey = computed(() => { return itemKeys.value[itemKeys.value.length - 1] })
+    const currentFocusItemIndex = computed(() => { return itemKeys.value.indexOf(currentFocusItem.value) })
+
+    const nextItemKey = computed(() => {
+        return currentFocusItem.value === lastItemKey.value
+            ? (wrap ? firstItemKey.value : null)
+            : itemKeys.value[currentFocusItemIndex.value + 1]
+    })
+
+    const prevItemKey = computed(() => {
+        return currentFocusItemIndex.value === 0
+            ? (wrap ? lastItemKey.value : null)
+            : itemKeys.value[currentFocusItemIndex.value - 1]
+    })
+
+    const navigateNextItem = (evt) => {
+        if (nextItemKey.value) {
+            evt.preventDefault();
+            const nextItemButton = itemSection.value.querySelector(`[id="${nextItemKey.value}"]`)
+            nextItemButton.click()
+        }
+    }
+
+    const navigatePrevItem = (evt) => {
+        if (prevItemKey.value) {
+            evt.preventDefault();
+            const prevItemButton = itemSection.value.querySelector(`[id="${prevItemKey.value}"]`)
+            prevItemButton.click()
+        }
+    }
+
+    const setInitActiveItem = () => {
+        if (location.hash && itemKeys.indexOf(location.hash.substr(1)) !== -1) {
+            activeItem.value = location.hash.substr(1)
+        } else if (initialItem) {
+            activeItem.value = initialItem
+        } else {
+            [activeItem.value] = items
+        }
+    }
+
+    const attachEventListeners = () => {
+        const firstItemEl = itemSection.value.querySelector(`[id="${firstItemKey.value}"]`)
+        const lastItemEl = itemSection.value.querySelector(`[id="${lastItemKey.value}"]`)
+
+        itemSection.value.addEventListener('click', event => {
+            event.preventDefault()
```

Comment:
For all the event handlers, only call `event.preventDefault()` if you're doing to do something, otherwise let it bubble.